perun-wui-profile/src/main/java/cz/metacentrum/perun/wui/profile/pages/settings/altPasswords/AltPasswordsView.java

```diff
@@ -0,0 +1,197 @@
+		getUiHandlers().navigateBack();
+	}
+
+	@UiHandler("closeModalButton")
+	public void closeModalButtonAction(ClickEvent event) {
+		generatedPasswordModal.hide();
+	}
+
+	@UiHandler("copyPasswordButton")
+	public void copyPasswordButtonAction(ClickEvent event) {
+		generatedPasswordBox.setFocus(true);
+		generatedPasswordBox.selectAll();
+		jSCopy();
+		generatedPasswordBox.setFocus(false);
+	}
+
+	@Inject
+	public AltPasswordsView(AltPasswordsViewUiBinder binder) {
+		initWidget(binder.createAndBindUi(this));
+		passwordBox.setWidth("100%");
+		addAltPasswordButton.setWidth("100%");
+		initTable(aPDataGrid);
+		generatedPasswordModal.addShownHandler(modalShownEvent -> generatedPasswordBox.setFocus(true));
+
+		passwordBox.addKeyUpHandler(keyUpEvent -> {
+			if (!passwordBox.getText().equals("") && !getUiHandlers().descriptionIsUsed(passwordBox.getText())) {
+				addAltPasswordButton.setEnabled(true);
+			} else {
+				addAltPasswordButton.setEnabled(false);
+			}
+		});
+	}
+
+	@Override
+	public void onError(PerunException e) {
+		invalidValueAlert.setVisible(true);
+		invalidValueAlert.setText(e.getMessage());
+		if (aPDataGrid.getRowCount() == 0) {
+			((PerunLoader) aPDataGrid.getEmptyTableWidget()).onEmpty();
+		}
+	}
+
+	@Override
+	public void setData(Map<String, JSONValue> data) {
+		invalidValueAlert.setVisible(false);
+		addAltPasswordButton.setEnabled(false);
+		if (data.isEmpty()) {
+			((PerunLoader) aPDataGrid.getEmptyTableWidget()).onEmpty();
+			aPDataGrid.setRowCount(0);
+		} else {
+			Map<String, String> tableData = new HashMap<>();
+			for (Map.Entry<String, JSONValue> entry : data.entrySet()) {
+				tableData.put(entry.getKey(), entry.getValue().toString());
+			}
+			aPDataGrid.setRowData(new ArrayList<>(tableData.entrySet()));
+		}
+
+	}
+
+	@Override
+	public void onLoadingStart() {
+		((PerunLoader) aPDataGrid.getEmptyTableWidget()).onLoading(translation.loadingUserData());
+	}
+
+	private void initTable(DataGrid<Map.Entry<String, String>> dataGrid) {
+
+		TextColumn<Map.Entry<String, String>> nameCol = new TextColumn<Map.Entry<String, String>>() {
+
+			@Override
+			public String getValue(Map.Entry<String, String> entry) {
+				return entry.getKey();
+			}
+		};
+
+		Column<Map.Entry<String, String>, String> deleteCol =
+				new Column<Map.Entry<String, String>, String>(new ButtonCell(ButtonType.DANGER)) {
+
+					@Override
+					public String getValue(Map.Entry<String, String> entry) {
+						return translation.delete();
+					}
+				};
+
+		deleteCol.setFieldUpdater((i, entry, s) -> getUiHandlers().deleteAltPassword(entry.getValue()));
+		PerunLoader pl = new PerunLoader();
+		pl.setEmptyMessage(translation.noPasswords());
+		pl.getElement().getStyle().setMarginTop(20, Style.Unit.PX);
+		dataGrid.setEmptyTableWidget(pl);
+		dataGrid.addColumn(nameCol, translation.description());
+		dataGrid.addColumn(deleteCol, "");
+		if (Window.Location.getHref().contains("locale=cs")) {
```

Comment:
That is not how we decide locale in GWT. Please see ``PerunConfiguration`` in ``perun-wui-core`` and look for ``getCurrentLocaleName()``.